orderbook/src/api/filter.rs

```diff
@@ -80,6 +88,41 @@ pub mod test_util {
         assert_eq!(response_orders, orderbook_orders);
     }
 
+    #[tokio::test]
+    async fn get_order_by_uid_() {
+        let orderbook = Arc::new(OrderBook::default());
+        let filter = get_order_by_uid(orderbook.clone());
+        let mut order_creation = OrderCreation::default();
+        order_creation.valid_to = u32::MAX;
+        order_creation.sign_self();
+        let order = orderbook.order_creation_to_order(order_creation).unwrap();
+        orderbook.add_order(order_creation).await.unwrap();
+        let response = request()
+            .path(&format!("/orders/{:}", order.order_meta_data.uid))
+            .method("GET")
+            .reply(&filter)
+            .await;
+        assert_eq!(response.status(), StatusCode::OK);
+        let response_orders: Order = serde_json::from_slice(response.body()).unwrap();
+        let orderbook_orders = orderbook.get_orders().await;
+        assert_eq!(response_orders, orderbook_orders[0]);
+    }
+    #[tokio::test]
+    async fn get_order_by_uid_for_non_existent_order_() {
+        let orderbook = Arc::new(OrderBook::default());
+        let filter = get_order_by_uid(orderbook.clone());
+        let mut order_creation = OrderCreation::default();
+        order_creation.valid_to = u32::MAX;
+        order_creation.sign_self();
+        let order = orderbook.order_creation_to_order(order_creation).unwrap();
+        let response = request()
+            .path(&format!("/orders/{:}", order.order_meta_data.uid))
```

Comment:
No need to create the full order here. You can create the uid directly like `let uid = OrderUid([u8; 56])` or add a derive for Default to OrderUid and use that. But if you prefer having the order here that's fine too.